nginx/nginx_test.go

```diff
@@ -859,6 +883,60 @@ func TestNginxIngressEntries(t *testing.T) {
 				"ssl_protocols TLSv1.2;",
 			},
 		},
+		{
+			"OpenTracing propagated the context",
+			opentracingConf,
+			[]controller.IngressEntry{
+				{
+					Host:           "chris.com",
+					Namespace:      "core",
+					Name:           "chris-ingress",
+					Path:           "/path",
+					ServiceAddress: "service",
+					ServicePort:    9090,
+				},
+			},
+			nil,
+			[]string{
+				"opentracing_propagate_context;",
+			},
+		},
+		{
+			"OpenTracing sets the operation name with path",
+			opentracingConf,
+			[]controller.IngressEntry{
+				{
+					Host:           "chris.com",
+					Namespace:      "core",
+					Name:           "chris-ingress",
+					Path:           "/path",
+					ServiceAddress: "service",
+					ServicePort:    9090,
+				},
+			},
+			nil,
+			[]string{
+				"opentracing_operation_name '$server_name /path/';",
+			},
+		},
+		{
+			"OpenTracing sets the operation name without path",
+			opentracingConf,
+			[]controller.IngressEntry{
+				{
+					Host:           "chris.com",
+					Namespace:      "core",
+					Name:           "chris-ingress",
+					Path:           "",
+					ServiceAddress: "service",
+					ServicePort:    9090,
+				},
+			},
+			nil,
+			[]string{
+				"opentracing_operation_name '$server_name /';",
+			},
+		},
```

Comment:
Should we add a check for none of the opentracing directives if they are not enabled?

Reply:
Already checked by "Check full ingress entry works"